src/resmom/linux/alps.c

```diff
@@ -5110,213 +5892,424 @@ alps_cancel_reservation(job *pjob)
-						while (tmp) {
-							if ((strcmp(basilversion, tmp)) == 0) {
-								/* Success!  ALPS and PBS speak BASIL 1.2 */
-								sprintf(log_buffer, "The basilversion is "
-									"set to %s", basilversion);
-								log_event(PBSEVENT_DEBUG,
-									PBS_EVENTCLASS_NODE,
-									LOG_DEBUG, __func__, log_buffer);
-								free(ver);
-								free_basil_response_data(brp);
-								return;
-							}
-							tmp = strtok(NULL, ",");
-						}
-						/* We didn't find "1.2" as a supported version  */
-						sprintf(log_buffer, "ALPS ENGINE query failed. "
-							"Supported BASIL versions returned: "
-							"'%s'", ver);
-						log_event(PBSEVENT_DEBUG3, PBS_EVENTCLASS_NODE,
-							LOG_NOTICE, __func__, log_buffer);
-					} else {
-						/* No memory */
-						sprintf(log_buffer, "ALPS ENGINE query failed.  No "
-							"memory");
-						log_event(PBSEVENT_ERROR, PBS_EVENTCLASS_NODE,
-							LOG_NOTICE, __func__, log_buffer);
-					}
-				} /* basil_support isn't in the response fall through
-				   * to try the next basil version
-				   */
-			} else {
-				/* wrong method in the response */
-				sprintf(log_buffer, "Wrong method, expected: %d but "
-					"got: %d", basil_method_query, brp->method);
-				log_event(PBSEVENT_DEBUG, PBS_EVENTCLASS_NODE,
-					LOG_DEBUG, __func__, log_buffer);
-			}
+	if (!brp) {
+		snprintf(log_buffer, sizeof(log_buffer),
+			 "Failed to switch %s ALPS reservation.", actionstring);
+		log_event(PBSEVENT_SYSTEM, PBS_EVENTCLASS_NODE, LOG_NOTICE,
+			  __func__, log_buffer);
+		return(-1);
+	}
+	if (*brp->error != '\0') {
+		/* A TRANSIENT error would mean the previous switch
+		 * method had not completed
+		 */
+		if (brp->error_flags & BASIL_ERR_TRANSIENT) {
+			free_basil_response_data(brp);
+			brp = NULL;
+			return (1);
 		} else {
-			/* There was an error in the BASIL response */
-			sprintf(log_buffer, "Error in BASIL response: %s", brp->error);
-			log_event(PBSEVENT_DEBUG, PBS_EVENTCLASS_NODE, LOG_DEBUG,
-				__func__, log_buffer);
+			free_basil_response_data(brp);
+			brp = NULL;
+			return(-1);
 		}
-	} else {
-		sprintf(log_buffer, "ALPS ENGINE query failed with BASIL "
-			"version %s.", basilversion);
-		/*
-		 * We log this a DEBUG3 because BASIL 1.2 may be too new
-		 * for this ALPS, and we'll try other BASIL versions next
-		 */
-		log_event(PBSEVENT_DEBUG3, PBS_EVENTCLASS_NODE,
-			LOG_NOTICE, __func__, log_buffer);
 	}
+	log_event(PBSEVENT_DEBUG2, PBS_EVENTCLASS_NODE, LOG_DEBUG,
+		__func__, "Made the ALPS SWITCH request.");
+	free_basil_response_data(brp);
+	return (0);
+}
 
-	if (ver) {
-		free(ver);
-		ver = NULL;
+/**
+ * @brief
+ * Confirm that an ALPS reservation has been switched to either "SUSPEND"
+ * or "RUN" status.
+ * Confirm that an ALPS reservation has successfully finished switching in/out.
+ *
+ * @retval 0 success
+ * @retval 1 transient error (retry)
+ * @retval -1 fatal error
+ */
+int
+alps_confirm_suspend_resume(job *pjob, basil_switch_action_t switchval)
```

Comment:
This function seems outside the scope of using the new BASIL 1.3/1.4 interface to fill in information for resources PBS already has. I don't think this code should be added.